constraint-solver/src/grouped_expression.rs

```diff
@@ -474,7 +475,7 @@ impl<T: FieldElement, V> RangeConstraintProvider<T, V> for NoRangeConstraints {
 impl<
         T: RuntimeConstant + Display + ExpressionConvertible<<T as RuntimeConstant>::FieldType, V>,
         V: Ord + Clone + Eq + Hash + Display,
-    > GroupedExpression<T, V>
```

Comment:
This is the central change, solving now only happens on constraints not expressions